contracts/v2/EIP3009.sol

```diff
@@ -68,20 +60,22 @@ abstract contract GasAbstraction is AbstractFiatTokenV2, EIP712Domain {
 
     /**
      * @notice Returns the state of an authorization
+     * @dev Nonces are randomly generated 32-byte data unique to the
+     * authorizer's address
      * @param authorizer    Authorizer's address
      * @param nonce         Nonce of the authorization
-     * @return Authorization state
+     * @return True if the nonce is used
      */
     function authorizationState(address authorizer, bytes32 nonce)
         external
         view
-        returns (AuthorizationState)
+        returns (bool)
```

Comment:
If a contract was compiled against the old version, what does this map to in terms of the enum? Does true get interpreted as `Used`? which would be nice

Reply:
The enum gets compiled to integer with possible values of `0`, `1`, and `2`. When read as "bool", the value of `0` gets interpreted as `false` and others get interpreted as `true`.

Reply:
Yeah that makes sense. I was thinking of the inverse - when a `bool` comes back but the caller reads it as an `AuthorizationState`. I think if `true` comes back, it should be as `1`, which looks like it would map to `AuthorizationState.Used`?

Reply:
@eztierney Yes that's correct:

```solidity
pragma solidity 0.6.12;

interface Legacy {
    enum AuthorizationState { Unused, Used, Canceled }

    function authorizationState() external view returns (AuthorizationState);
}

contract New {
    bool public state = true;

    function authorizationState() external view returns (bool) {
        return state;
    }
    
    function legacyAuthorizationState() external view returns (Legacy.AuthorizationState) {
        Legacy.AuthorizationState x = Legacy(address(this)).authorizationState();
        
        return x; // returns 1 (Legacy.AuthorizationState.Used)
    }
}
```